src/Raven.Server/Documents/Replication/ReplicationLoader.cs

```diff
@@ -402,14 +412,14 @@ private void HandleTopologyChange(DatabaseRecord newRecord, out List<OutgoingRep
             lock (_locker)
             {
                 instancesToDispose = new List<OutgoingReplicationHandler>();
-                if (newRecord == null)
+                if (newRecord == null || DatabaseTopology.PartOfCluster == false)
```

Comment:
You can have multiple servers in the same process. 
If you want to make this a global per server value, put in on the server instance.